d2compiler/compile.go

```diff
@@ -761,6 +794,19 @@ func (c *compiler) validateNear(g *d2graph.Graph) {
 	}
 }
 
+func (c *compiler) validateEdges(g *d2graph.Graph) {
+	for _, edge := range g.Edges {
+		if gd := edge.Src.ClosestGridDiagram(); gd != nil {
+			c.errorf(edge.GetAstEdge(), "edge %#v cannot enter grid diagram %#v", d2format.Format(edge.GetAstEdge()), gd.AbsID())
```

Comment:
"enter" can be confused as a special term. `cannot connect children of grid diagrams currently`